optout/src/main/java/gov/cms/ab2d/optout/OptOutProcessing.java

```diff
@@ -0,0 +1,182 @@
+        try {
+            String mbi = information.substring(MBI_INDEX_START, MBI_INDEX_END).trim();
+            Timestamp effectiveDate = convertToDate(information.substring(EFFECTIVE_DATE_INDEX_START, EFFECTIVE_DATE_INDEX_END));
+            boolean optOutFlag = (information.charAt(OPTOUT_FLAG_INDEX) == 'Y');
+
+            OptOutInformation optOutInformation = new OptOutInformation(mbi, effectiveDate, optOutFlag, lineNumber, information);
+            // The file line was parsed successfully
+            optOutResultMap.put(lineNumber, new OptOutResult(optOutInformation, RecordStatus.ACCEPTED, ReasonCode.ACCEPTED));
+            return Optional.of(optOutInformation);
+        } catch (NumberFormatException | StringIndexOutOfBoundsException | ParseException ex) {
+            logger.log("Lambda can not parse the line: " + lineNumber);
+            // The file line was parsed with an error
+            optOutResultMap.put(lineNumber, new OptOutResult(new OptOutInformation(lineNumber, information), RecordStatus.REJECTED, ReasonCode.PARSE_ERROR));
+        }
+        return Optional.empty();
+    }
+
+    public void updateOptOut(OptOutInformation optOutInformation, Connection dbConnection) {
+        try (PreparedStatement statement = dbConnection.prepareStatement(UPDATE_STATEMENT)) {
+            logger.log("Mbi: " + optOutInformation.getMbi() + ", OptOut Flag: " + optOutInformation.isOptOut());
+            prepareInsert(optOutInformation, statement);
+            statement.execute();
+        } catch (SQLException ex) {
+            optOutResultMap.put(optOutInformation.getLineNumber(),
+                    new OptOutResult(
+                            new OptOutInformation(optOutInformation.getLineNumber(), optOutInformation.getText()),
+                            RecordStatus.REJECTED,
+                            ReasonCode.INSERT_ERROR));
+            logger.log(ex.getMessage());
+        }
+    }
+
+    public String createResponseContent() {
+        StringBuilder responseContent = new StringBuilder();
+        for (Map.Entry<Long, OptOutResult> optOutResult : optOutResultMap.entrySet()) {
+            OptOutResult line = optOutResult.getValue();
+            String text = line.getOptOutInformation().getText();
+            // First and last lines don't contain optout data and are written as is
+            if (optOutResult.getKey() == 0 || optOutResult.getKey() == optOutResultMap.size() - 1) {
+              responseContent.append(text);
+            } else {
+                StringBuilder result = new StringBuilder(text);
+                // Adding spaces to the end of a string to achieve the RecordStatus position index
+                if (text.length() < DEFAULT_LINE_LENGTH)
+                    result.append(" ".repeat(Math.max(0, DEFAULT_LINE_LENGTH - text.length())));
+
+                result.append(String.format(RECORD_STATUS_PATTERN, line.getRecordStatus()));
+                result.append(line.getReasonCode());
```

Comment:
Retries before we append reason code.. especially when it fails as we discussed

Reply:
[AB2D-5799](https://jira.cms.gov/browse/AB2D-5799) Handle Error Cases